internal/pkg/manifest/validate.go

```diff
@@ -1856,6 +1860,62 @@ func populateALBPortsAndValidate(containerNameFor map[uint16]string, opts valida
 	return nil
 }
 
+func populateNLBPortsAndValidate(containerNameFor map[uint16]string, opts validateExposedPortsOpts) error {
+	if opts.nlb == nil {
+		return nil
+	}
+	nlb := opts.nlb
+	if nlb.Port == nil {
+		return nil
+	}
+	nlbPort, protocol, err := ParsePortMapping(nlb.Port)
+	if err != nil {
+		return err
+	}
+	if protocol != nil {
+		protocolVal := aws.StringValue(protocol)
+		var isValidProtocol bool
+		for _, valid := range nlbValidProtocols {
+			if strings.EqualFold(protocolVal, valid) {
+				isValidProtocol = true
+				break
+			}
+		}
+		if !isValidProtocol {
+			return fmt.Errorf(`invalid protocol %s; valid protocols include %s`, protocolVal, english.WordSeries(nlbValidProtocols, "and"))
+		}
+	}
+	port, err := strconv.ParseUint(aws.StringValue(nlbPort), 10, 16)
+	if err != nil {
+		return err
+	}
+	containerPort := uint16(port)
+	if nlb.TargetPort != nil {
+		containerPort = uint16(aws.IntValue(nlb.TargetPort))
+	}
+
+	containerName := opts.mainContainerName
+	existingContainerName := containerNameFor[containerPort]
+
+	if existingContainerName != "" {
+		containerName = existingContainerName
```

Comment:
can we remove this line?

Reply:
May I ask why do we want to remove this line?

Reply:
because `existingContainerName` would be a better name and easier to understand than just `containerName`